framework/src/streaming/lib/msk/kafka-api.ts

```diff
@@ -96,41 +101,51 @@ export class KafkaApi extends TrackedConstruct {
 
     if (props.clientAuthentication.tlsProps?.certificateAuthorities) {
 
-      const mskAclProvider = mskAclAdminProviderSetup(
-        this,
-        this.removalPolicy,
-        props.vpc,
-        props.subnets || props.vpc.selectSubnets({ subnetType: SubnetType.PRIVATE_WITH_EGRESS }),
-        props.brokerSecurityGroup,
-        props.clusterArn,
-        props.certficateSecret!,
-        props.mtlsHandlerRole,
-      );
-
-      this.mskAclServiceToken = mskAclProvider.serviceToken;
-      this.mskAclRole = mskAclProvider.onEventHandlerRole;
-      this.mskAclLogGroup = mskAclProvider.onEventHandlerLogGroup;
-      this.mskAclFunction = mskAclProvider.onEventHandlerFunction;
-      this.mskAclSecurityGroup = mskAclProvider.securityGroups;
+      if (props.serviceToken) {
+        this.mskAclServiceToken = this.serviceToken = props.serviceToken;
```

Comment:
can you please not to this? Only assign a value to `this.serviceToken` if we create it by the construct. If its passed by the user then we do not need to expose it since the user already knows about it.